pkg/preemptiontoleration/preemption_toleration_test.go

```diff
@@ -0,0 +1,241 @@
+package preemptiontoleration
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/scheduler-plugins/pkg/apis/config"
+	"sigs.k8s.io/scheduler-plugins/pkg/apis/config/v1beta1"
+	"sigs.k8s.io/yaml"
+
+	v1 "k8s.io/api/core/v1"
+	schedulingv1 "k8s.io/api/scheduling/v1"
+
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/cache"
+	st "k8s.io/kubernetes/pkg/scheduler/testing"
+)
+
+func TestGetCompletedPreemptionToleration(t *testing.T) {
+	tests := []struct {
+		name          string
+		priorityClass *schedulingv1.PriorityClass
+		want          *config.PreemptionToleration
+	}{
+		{
+			name:          "nil just returns nil",
+			priorityClass: makePriorityClass(t, samplePriority, nil),
+			want:          nil,
+		},
+		{
+			name:          "empty preemption toleration completes MinimumPreemptablePriority",
+			priorityClass: makePriorityClass(t, samplePriority, &v1beta1.PreemptionToleration{}),
+			want: &config.PreemptionToleration{
+				MinimumPreemptablePriority: pointer.Int32Ptr(samplePriority + 1),
+			},
+		},
+		{
+			name: "preemption toleration does not complete TolerationSeconds",
+			priorityClass: makePriorityClass(t, samplePriority, &v1beta1.PreemptionToleration{
+				MinimumPreemptablePriority: pointer.Int32Ptr(samplePriority + 10),
+			}),
+			want: &config.PreemptionToleration{
+				MinimumPreemptablePriority: pointer.Int32Ptr(samplePriority + 10),
```

Comment:
`cmp.Diff(x, y)` used by the assertion should assert it because `nil` would be the default value of the struct.  But, I agree explicit expected declaration would be better.

Reply:
Thank you for the detailed comments and updates!